### PR TITLE
Support building development lantern on Windows

### PR DESCRIPTION
**Pull request by oxtoacart**
The packaging stuff won't work yet, but at least I can build the `lantern` target and run Lantern.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -200,13 +200,18 @@ $(RESOURCES_DOT_GO): $(NPM)
...
+	rm -f bin/tarfs && \
 	go install github.com/getlantern/tarfs/tarfs && \
+	mv bin/tarfs.exe bin/tarfs ; \
 	echo "// +build !stub" > $$DEST && \
 	echo " " >> $$DEST && \
-	tarfs -pkg ui $$DIST >> $$DEST && \
+	bin/tarfs -pkg ui $$DIST >> $$DEST
+
+# Generates a syso file that embeds an icon for the Windows executable
+generate-windows-icon:
```
Comment (oxtoacart) [inline Makefile]: The syso is now back in git so I don't need to run this on Windows, which is good because it doesn't run.

**Comment by xiam** [comment]
Gonna take this one, looks like we have a conflict @oxtoacart, could you rebase and push again?

**Comment by oxtoacart** [comment]
@xiam Merged

**Comment by oxtoacart** [comment]
@xiam I mean, I reverse merged to resolve the conflict :)

**Comment by xiam** [comment]
Hey @oxtoacart, looks good, just made a few changes. What are the name changes required for?

**Comment by oxtoacart** [comment]
@xiam By "name changes" do you mean all the moved files in lantern-ui?  With lantern-ui in the original location, the directory structure is too deep for Windows which has a maximum path length of 260.

**Comment by xiam** [comment]
Yeah, that's it. Hmm, not sure if that justifies changing the path for the
whole project. Isn't it possible to use soft links or equivalent? I'll take
a look too trying to find another solution.

**Comment by oxtoacart** [comment]
@xiam I only moved lantern-ui.  It's actually kind of strange to have it in the GOPATH in the first place, and I used git mv to move it, so it shouldn't cause any merge issues.

**Comment by myleshorton** [comment]
Yeah I agree @oxtoacart it's a little weird to have lantern-ui in the GOPATH.

**Comment by xiam** [comment]
Ok then @myleshorton & @oxtoacart, I think that having our source code on `src` makes more sense, but if you both agree let's leave it like that. Otherwise the build works for me.

**Comment by oxtoacart** [comment]
@xiam thanks for the review and improving the Makefile!

**Comment by myleshorton** [comment]
Nice @xiam!!

**Pull request merged**